### DISCUSSION TITLE
Measure DC current

### DISCUSSION THREAD

**Discussion by eufrizz** [tinymovr/Tinymovr#210] [Q&A - Tinymovr]
Hi Yannis,

If I wanted to measure the instantaneous DC current being used by the motor (as a power meter might read), how could I do this? Would it just be a matter of adding the currents on each phase or is there more to it? I'm a little confused by what's going on in the ADC_UpdateMeasurements function and can't seem to find any info on current readings in the documentation.

Thanks!

**Comment by yconst** [comment]
(I converted this to a discussion as I think it is mostly a question, feel free to raise another issue if you think there is an issue with ADC measurements)

I'm not sure what you mean by DC current. Is it the current in the dq frame? If so, it can be measured directly. Only q is needed as d is normally zero. Nonetheless, with the comms layer revision we'll add access to the d component as well.

In addition, you can also find out the measured phase currents.

If there is a specific question you have regarding ADC measurements, feel free to drop it here!

**Reply by eufrizz** [reply]
(Yes I should have made it a discussion, thanks for converting it! No issue with the ADC measurements)
I have the motor on a power supply, and can read the current it is drawing. It is this DC current that I would like to measure on the ESC - nothing to do with the rotational frame of the motor. Comparing the power supply current reading to the Iq reading, I notice the relationship seems nonlinear (so it is not just a scaling error). Also, Iq is negative when the motor is spinning backwards. I'm just interested in seeing the DC current and power drawn by the ESC to see e.g. what the motor efficiency is/what current a battery would have to supply/what current the motor leads would need to be rated for.
Apologies for my lack of knowledge on FOC, I hope I've explained myself clearly enough!

**Reply by yconst** [reply]
Ok now I understand, the short answer is that the current supplied by the power supply != the current flowing in the motor phases. 

The motor controller is a current regulator, meaning that it will adjust the voltage across the motor phases to achieve the desired current. Because the motor impedance is generally small, the desired current can be achieved with low phase voltage, compared to the input voltage. This is a bit like how a buck converter works, although there you have voltage being regulated instead of current.

Most motor controllers don't have a way of measuring input current, as this would require additional components and introduce additional inefficiencies.

A very good overview of the topic is available in the following blog post: https://things-in-motion.blogspot.com/2018/12/how-to-select-right-power-source-for.html
